File: scripts/problems/binary-search/boilerplate-full.js
const lines = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
let _idx = 0;
const nextLine = () => lines[_idx++];

// ── USER CODE START ──
{{USER_CODE}}
// ── USER CODE END ──

const n_nums = parseInt(nextLine());
const nums = nextLine().split(" ").slice(0, n_nums).map(Number);
const target = parseInt(nextLine());

const result = search(nums, target);
console.log(result);

File: scripts/problems/binary-search/boilerplate-full.rs
use std::io::{self, BufRead};

fn next_line() -> String {
    let stdin = io::stdin();
    let mut lines = stdin.lock().lines();
    lines.next().unwrap().unwrap()
}

// ── USER CODE START ──
{{USER_CODE}}
// ── USER CODE END ──

fn main() {
    let _n_nums: usize = next_line().trim().parse().unwrap();
let nums: Vec<i64> = next_line().split_whitespace().map(|x| x.parse().unwrap()).collect();
    let target: i64 = next_line().trim().parse().unwrap();

    let result = search(nums, target);
    println!("{}", result);
}